### PR TITLE
vendor dropdown component added

### PR DESCRIPTION
**Pull request by rishali**

### CODE REVIEW

**Comment by rishali** [comment]
@sahuaccolite @prashantaccolite

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
-  styleUrls: ['./app.component.scss']
+  styleUrls: ['./app.component.scss'],
 })
 export class AppComponent {
-  title = 'TFR-Management';
+  vendors: Vendor[] = [];
+
+   title = 'TFR-Management';
+  constructor(private service: ApiserviceService) {}
+  ngOnInit() {
```
Comment (prasadkorrapati) [inline src/app/app.component.ts:15]: This does not need here. the component who is using this  vendor component may do the implementation

File: src/app/app.component.html [hunk 2]
```diff
@@ -32,7 +32,8 @@
       </mat-sidenav>
       <mat-sidenav-content fxFlexFill>
         TFR Details 
+      <app-vendors [vendors]="vendors" (onSelected)="onVendorSelected($event)"
```
Comment (prasadkorrapati) [inline src/app/app.component.html:35]: This is not required here now. TFR details component will use this later

File: src/app/app.module.ts [hunk 3]
```diff
@@ -5,18 +5,22 @@ import { AppRoutingModule } from './app-routing.module';
...
-    BrowserModule,
-    AppRoutingModule,
-    BrowserAnimationsModule,
-    CoreMaterialModule
-  ],
-  providers: [],
-  bootstrap: [AppComponent]
+    declarations: [
+        AppComponent,
+        VendorsComponent    ],
```
Comment (prasadkorrapati) [inline src/app/app.module.ts:15]: declarations: [
        AppComponent,
        VendorsComponent
],

properly format this code above

File: src/app/core-modules/core-material/core-material.module.ts [hunk 4]
```diff
@@ -18,17 +18,19 @@ import {
 import { BrowserModule } from '@angular/platform-browser';
 import { FormsModule } from '@angular/forms';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import {MatSelectModule} from '@angular/material/select';
+
+const materialModules = [MatIconModule, MatButtonModule, MatSidenavModule, MatToolbarModule,
```
Comment (prasadkorrapati) [inline src/app/core-modules/core-material/core-material.module.ts:23]: const materialModules = [
  MatIconModule,
  MatButtonModule,
  MatSidenavModule,
  MatToolbarModule,
  MatSelectModule,
  FlexLayoutModule,
];
Indent properly as above

File: src/app/core-modules/core-material/core-material.module.ts [hunk 5]
```diff
@@ -18,17 +18,19 @@ import {
...
+  MatSelectModule,FlexLayoutModule]
 
 @NgModule({
   declarations: [],
-  imports:      [
-    BrowserModule, FormsModule, FlexLayoutModule, BrowserAnimationsModule,
-    MatIconModule, MatButtonModule, MatSidenavModule, MatToolbarModule,
-    CommonModule
-  ],
+  imports:  [BrowserModule, FormsModule, BrowserAnimationsModule,CommonModule,
```
Comment (prasadkorrapati) [inline src/app/core-modules/core-material/core-material.module.ts:28]: imports: [
  BrowserModule,
  FormsModule,
  BrowserAnimationsModule,
  CommonModule,
  materialModules,
];
format code as above

File: src/app/core-modules/core-material/core-material.module.ts [hunk 6]
```diff
@@ -18,17 +18,19 @@ import {
...
+    materialModules],
   exports:[
-    FlexLayoutModule,
```
Comment (prasadkorrapati) [inline src/app/core-modules/core-material/core-material.module.ts:30]: exports: [
  BrowserModule,
  FormsModule,
  BrowserAnimationsModule,
  CommonModule,
  materialModules,
];

File: src/app/service/apiservice.service.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+import { Vendor } from '../types/types';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ApiserviceService {
+  Vendors: Vendor[] = require('../../assets/json/vendors.json');
+
+  constructor(private httpClient: HttpClient) {
+    console.log('Vendors ', this.Vendors);
```
Comment (prasadkorrapati) [inline src/app/service/apiservice.service.ts:13]: you can remove this console.log

File: src/app/service/apiservice.service.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  }
+  getVendorData(): Observable<Vendor[]> {
```
Comment (prasadkorrapati) [inline src/app/service/apiservice.service.ts:15]: looks good.

File: src/app/types/types.d.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
+export interface Vendor {
```
Comment (prasadkorrapati) [inline src/app/types/types.d.ts:1]: Looks good.

File: src/assets/json/vendors.json [hunk 10]
```diff
@@ -0,0 +1,12 @@
+[{
```
Comment (prasadkorrapati) [inline src/assets/json/vendors.json:1]: Looks good.

**Comment by rishali** [comment]
Vendor commit update

File: src/app/app.component.html [hunk 11]
```diff
@@ -32,7 +32,8 @@
       </mat-sidenav>
```
Comment (rishali) [inline src/app/app.component.html:32]: testing

**Review by prasadkorrapati — CHANGES_REQUESTED** [review]

File: src/app/app.component.html [hunk 12]
```diff
@@ -1,38 +1,40 @@
...
-          <mat-icon>menu</mat-icon>
-        </button>
-      </div>
-  
-      <a mat-button class="acc-tfr" routerLink="/">
-        <span>TFR Managment</span>
-      </a>
-      <span class="example-spacer"></span>
-      <div fxShow="true" fxHide.lt-md="true">
+<div style="height: 100vh">
```
Comment (prasadkorrapati) [inline src/app/app.component.html:1]: looks good

File: src/app/components/vendors/vendors.component.html [hunk 13]
```diff
@@ -0,0 +1,11 @@
+<mat-form-field appearance="fill">
```
Comment (prasadkorrapati) [inline src/app/components/vendors/vendors.component.html:1]: looks good

File: src/app/core-modules/core-material/core-material.module.ts [hunk 14]
```diff
@@ -1,34 +1,35 @@
...
-  MatToolbarModule
-} from '@angular/material/toolbar';
+import { MatSidenavModule } from '@angular/material/sidenav';
+import { MatToolbarModule } from '@angular/material/toolbar';
 import { BrowserModule } from '@angular/platform-browser';
 import { FormsModule } from '@angular/forms';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { MatSelectModule } from '@angular/material/select';
+
+const materialModules = [
```
Comment (prasadkorrapati) [inline src/app/core-modules/core-material/core-material.module.ts:15]: looks good

File: src/app/service/apiservice.service.ts [hunk 15]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { of, Observable } from 'rxjs';
+import { Vendor } from '../types/types';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ApiserviceService {
```
Comment (prasadkorrapati) [inline src/app/service/apiservice.service.ts]: Change the service name to ApiService

**Review by prasadkorrapati — APPROVED** [review]
Looks good.

**Comment by rishali** [comment]
@sahuaccolite

**Comment by sahuaccolite** [comment]
Test

File: src/app/components/charts/charts.service.ts [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+import { IStatus } from 'src/app/Interface/Status.Interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ChartsService {
+  //private StatusUrl = 'http://localhost:4200/projects/statusCount';
+
+  constructor(private http: HttpClient) {
+    this.readTfrStatusData().subscribe((data) => {});
```
Comment (sahuaccolite) [inline src/app/components/charts/charts.service.ts:13]: Why this needed ? please remove this

File: src/app/components/charts/charts.service.ts [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  // getStatusData(): Observable<any> {
+  //   return this.http.get(this.StatusUrl);
+  // }
+
+  tfrStatus: IStatus[] = require('../../../assets/json/status.json');
```
Comment (sahuaccolite) [inline src/app/components/charts/charts.service.ts:20]: As already  suggested we not need require here here we just need path and call get directly let me know if anything needs to be dicsussed

**Review by sahuaccolite — APPROVED** [review]

**Review by angularavengers — APPROVED** [review]

**Comment by sahuaccolite** [comment]
Merged

**Comment by sahuaccolite** [comment]
Merged

**Comment by sahuaccolite** [comment]
Merged

**Comment by sahuaccolite** [comment]
Merged

**Comment by sahuaccolite** [comment]
Approved

**Review by sahuaccolite — APPROVED** [review]

**Pull request merged**